cmd/parca-agent/main.go

```diff
@@ -169,29 +175,38 @@ func run(logger log.Logger, reg *prometheus.Registry, flags flags) error {
 		"arch", goArch,
 	)
 
-	var (
-		ctx             = context.Background()
-		debugInfoClient = debuginfo.NewNoopClient()
+	profileStoreClient := agent.NewNoopProfileStoreClient()
+	debugInfoClient := debuginfo.NewNoopClient()
 
-		g             okrun.Group
-		profileWriter profiler.ProfileWriter
-	)
 	if len(flags.RemoteStoreAddress) > 0 {
 		conn, err := grpcConn(reg, flags)
 		if err != nil {
 			return err
 		}
 		defer conn.Close()
 
-		profileStoreClient := profilestorepb.NewProfileStoreServiceClient(conn)
+		profileStoreClient = profilestorepb.NewProfileStoreServiceClient(conn)
 		if !flags.RemoteStoreDebugInfoUploadDisable {
 			debugInfoClient = parcadebuginfo.NewDebugInfoClient(conn)
 		} else {
 			level.Info(logger).Log("msg", "debug information collection is disabled")
 		}
+	}
+
+	var (
+		ctx = context.Background()
+
+		g                okrun.Group
+		batchWriteClient = agent.NewBatchWriteClient(logger, profileStoreClient, flags.RemoteStoreBatchWriteInterval)
+		profileListener  = agent.NewMatchingProfileListener(logger, batchWriteClient)
+		profileWriter    profiler.ProfileWriter
+	)
 
-		batchWriteClient := agent.NewBatchWriteClient(logger, profileStoreClient, flags.RemoteStoreBatchWriteInterval)
-		profileWriter = profiler.NewRemoteProfileWriter(agent.NewMatchingProfileListener(logger, batchWriteClient))
+	if flags.LocalStoreDirectory != "" {
```

Comment:
I think we should prioritize the remote writer over the local one. Or even better, if both are specified, we should have a writer to write both of them. We don't necessarily need to handle it in this PR, however, it would be great.